src/components/reviews/Reviews.jsx

```diff
@@ -1,73 +1,111 @@
 import React from 'react';
 import axios from 'axios';
 import API_KEY from '../../config.js';
+import ReviewsList from './ReviewsList.jsx';
 
 class Reviews extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       reviewsList: [],
       filteredReviewsList: [],
-      filter: () => true // Default "filter": displays all reviews.
+      filter: () => true, // Default "filter": displays all reviews.
+      reviewsPerLoad: 2,
+      reviewCount: 0,
+      moreToLoad: true,
+      isLoading: false
     };
-    this.loadReviews = this.loadReviews.bind(this);
+    this.loadMoreReviews = this.loadMoreReviews.bind(this);
     this.setFilter = this.setFilter.bind(this);
   }
 
   componentDidMount() {
-    this.loadReviews(1, 2);
+    this.fetchReviews(1, this.state.reviewsPerLoad);
   }
 
-  loadReviews(page, count) {
-    const intializationConfig = {
-      method: 'get',
-      url: 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-nyc/products',
-      headers: {
-        Authorization: API_KEY,
-      },
-      params: {
-        page,
-        count,
-        'product_id': this.props.productId,
-        'sort': 'newest'
-      }
-    };
+  fetchReviews(page, count, reviewsList = [], callback = null) {
+    const { productId } = this.props;
+    const { isLoading } = this.state;
+    if (productId && !isLoading) {
+      const intializationConfig = {
+        method: 'get',
+        url: 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-nyc/reviews',
+        headers: {
+          Authorization: API_KEY,
+        },
+        params: {
+          page,
+          count,
+          'sort': 'newest',
+          'product_id': this.props.productId
+        }
+      };
+
+      this.setState({ isLoading: true });
```

Comment:
I was wondering why you need 'isLoading' state in this feature? 🤔

Reply:
I was using it to make sure that the fetchReviews function can't be called if there's another call still waiting for a response from the server. I'm not sure it's still an issue but at one point it actually broke my widget if that happened.